### PR TITLE
Facia; Press on demand

### PR DESCRIPTION
**Pull request by janua**
Without `SQS`, there is no way to trigger a press in `facia-press`, and it can continue it's work without `SQS`.

This gives us the ability to press on demand.

In case you are wondering; what `SQS` delivers is just paths, such as `"uk"` or `"au/culture"`.

@robertberry @gklopper @stephanfowler

### CODE REVIEW

File: facia-press/conf/routes [hunk 1]
```diff
@@ -2,4 +2,7 @@ GET        /_healthcheck        conf.HealthCheck.healthcheck()
 GET        /                    controllers.Application.index
 GET        /debug/config        controllers.Application.showCurrentConfig
 
-GET        /pressed/live/*id      controllers.Application.generateLivePressedFor(id)
+GET        /pressed/live/*id    controllers.Application.generateLivePressedFor(id)
+
+GET        /press/live/*path    controllers.Application.pressLiveForPath(path)
+GET        /press/draft/*path   controllers.Application.pressDraftForPath(path)
```
Comment (stephanfowler) [inline facia-press/conf/routes]: Feels wrong for GET to have a side effect. But I guess it's acceptable as a switchable-off emergency URL that needs to be easy to hit...

**Comment by stephanfowler** [comment]
:+1:

File: facia-press/conf/routes [hunk 2]
```diff
@@ -2,4 +2,7 @@ GET        /_healthcheck        conf.HealthCheck.healthcheck()
 GET        /                    controllers.Application.index
 GET        /debug/config        controllers.Application.showCurrentConfig
 
-GET        /pressed/live/*id      controllers.Application.generateLivePressedFor(id)
+GET        /pressed/live/*id    controllers.Application.generateLivePressedFor(id)
```
Comment (gklopper) [inline facia-press/conf/routes]: is this ID the path of the URL?

Comment (janua) [inline facia-press/conf/routes]: That endpoint already existed to show what will be output for a given path. Should probably be `path`.

File: facia-press/conf/routes (hunk 1 above)
Comment (janua) [inline facia-press/conf/routes]: @stephanfowler Yes true. Let's make it `OPTIONS`

File: facia-press/app/controllers/Application.scala [hunk 3]
```diff
@@ -20,4 +24,21 @@ object Application extends Controller with ExecutionContexts {
     FrontPress.pressLiveByPathId(path).map(Json.stringify(_))
       .map(Ok.apply(_))
       .map(NoCache.apply)}
+
+  private def handlePressRequest(path: String, liveOrDraft: String)(f: (String) => Future[_]): Future[Result] =
+    if (FaciaPressOnDemand.isSwitchedOn) {
+      val stage = Configuration.facia.stage.toUpperCase
+      f(path)
```
Comment (gklopper) [inline facia-press/app/controllers/Application.scala]: What is `f()` - is it just the same as `Future.successful`?

If so then just make these non async.

Comment (janua) [inline facia-press/app/controllers/Application.scala]: No, `f` is returning a real future.

**Comment by gklopper** [comment]
:+1: 

But we will need to leave this switched off for many months at a time and then hope it still works when we have a problem.

Combine this with the fact that there is not one test for it and it feels a bit like [Generals fighting the last war](http://www.quora.com/What-are-some-good-examples-of-Generals-fighting-the-last-war)

**Comment by janua** [comment]
It'll work. For if it doesn't work then the website doesn't work.

**Pull request merged**